homeassistant/components/hydrawise/config_flow.py

```diff
@@ -22,25 +23,61 @@ class HydrawiseConfigFlow(ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Hydrawise."""
 
     VERSION = 1
-
-    async def _create_entry(
-        self, api_key: str, *, on_failure: Callable[[str], ConfigFlowResult]
+    MINOR_VERSION = 2
+
+    def __init__(self) -> None:
+        """Construct a ConfigFlow."""
+        self.reauth_entry: ConfigEntry | None = None
+
+    async def _create_or_update_entry(
+        self,
+        api_key: str | None = None,
+        username: str | None = None,
+        password: str | None = None,
```

Comment:
Why does this method have default values for its parameters?

Reply:
It is called from two different places. Once with the api_key, and once with the username and password.

Reply:
Yes, I see that, I'm just not a big fan of default values in cases like this.
I guess it's a matter of personal taste and it's OK like this.